src/plugins/graph/components/attribute-label.tsx

```diff
@@ -51,122 +62,84 @@ export const AttributeLabel = observer(
     }, [dataConfiguration, graphModel.plotType, place]);
 
     const getLabel = useCallback(() => {
+      // If we're using the CLUE legend, use a customizable label
+      if (defaultSeriesLegend) {
+        if (place === "left") {
+          return graphModel.yAttributeLabel;
+        } else {
+          return graphModel.xAttributeLabel;
+        }
+      }
       if (useClickHereCue) {
         // empty axis shows the default axis label (if configured) or the click here prompt
         return defaultAxisLabels?.[place] || t('DG.AxisView.emptyGraphCue');
       }
       const attrIDs = getAttributeIDs();
       return attrIDs.map(anID => dataset?.attrFromID(anID)?.name)
         .filter(aName => aName !== '').join(', ');
-    }, [dataset, defaultAxisLabels, getAttributeIDs, place, useClickHereCue]);
+    }, [dataset, defaultAxisLabels, defaultSeriesLegend, getAttributeIDs, graphModel, place, useClickHereCue]);
 
-    const refreshAxisTitle = useCallback(() => {
```

Comment:
I am a little worried about the several effects that were removed from this file. Are you confident they are not needed any more in the "codap legends" case?  I didn't see any problems but thought it was worth double checking.

Reply:
I'm not totally confident, but I'm pretty confident that all of these useEffects and reactions were needed because the labels were being drawn with d3. This work renders the labels with react, so they should be updated automatically, instead of having to be removed and redrawn when one of a specified list of things happens. Playing around with the old legend style in the moth unit, everything seems to work as expected. Maybe we should just make a point of calling this out to QA?

Reply:
Sounds good.